src/lib.rs

```diff
@@ -293,13 +293,19 @@ impl Mesh {
             start,
         );
 
-        loop {
+        // Limit search to avoid an infinite loop.
+        for _ in 0..self.polygons.len() * 1000 {
             match search_instance.next() {
                 InstanceStep::Found(path) => return Some(path),
                 InstanceStep::NotFound => return None,
                 InstanceStep::Continue => (),
             }
         }
+
+        // The above should always return a path, or find no path exists.
+        // In this case, the mesh is faulty in some way.
+        eprintln!("Mesh::path: search failed from {from} to {to}");
```

Comment:
Could you use a logger instead of `eprintln!`? The other prints are behind a feature meant to debug the implementation, this could be useful to actual users of the library

Reply:
Sure, should I use `tracing` or `log`? And if `tracing` should it be behind the `tracing` feature?

Reply:
`log` please!

Reply:
Updated. The log looks like this:
```
2024-01-18T10:46:36.453377Z ERROR log: Search from [58.245605, 74.68821] to [55.11034, 17.506596] failed. Please check the mesh is valid as this should not happen.
```